src/store/search.js

```diff
@@ -113,7 +117,11 @@ export const getters = {
     if (state.query.mature === true) {
       params.mature = true
     }
-    return params
+    if (!toCheckIfQueryisNull) {
+      return params
+    } else {
+      return redirect('/')
```

Comment:
You cannot called `redirect()` from a Vuex store. That function is not defined in this context.

Reply:
hey, thanks for your reply. 

So I am a complete beginner and I have no clue about how to approach it. However, I would still like to try for some time. 
Maybe If you could please give me the names of the file where changes have to be made I can have a starting point. 

Because from what I understand the `src/middleware/middleware.js` file has to be changed but I am not sure how because It is setting the relative URL path.

```
 * Currently, two event types are used:
 * - `resize` sends the height of the window (see `src/mixins/iframe-height.js`)
 * - `urlChange` sends the relative path of the URL on every URL change.
 */
export default function ({ query, route, $pinia }) {
  const navStore = useNavStore($pinia)

  if ('embedded' in query) {
    navStore.setIsEmbedded(query.embedded === 'true')
  }
  if (process.client) {
    sendWindowMessage({
      type: 'urlChange',
      value: { path: route.fullPath, title: document.title },
    })
  }

  if (process.client && navStore.isReferredFromCc) {
    navStore.setIsReferredFromCc(false)
  }
}


```

Reply:
In that function (or preferably a new, specific, single purpose middleware function) you can destructure `redirect` from the same argument that `$pinia`, `query` and `route` are destructured from. Then just inspect the `route` to see if it matches the `search` route. The `route` object will also have a `query` property that contains the list of URL query params.

https://router.vuejs.org/api/#query